app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,19 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  skip_before_action :authenticate_user!
+  include Devise::Controllers::Rememberable
+
+  def github
+    @user = User::AuthenticateFromOmniauth.(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      remember_me(@user)
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session["devise.github_data"] = request.env["omniauth.auth"].except(:extra)
+
+      set_flash_message(:alert, :failure, kind: "GitHub") if is_navigational_format?
```

Comment:
I'd like us to keep flash things for auth in their own "namespace". So can you change this to `auth_alert` pls?

```suggestion
      set_flash_message(:auth_alert, :failure, kind: "GitHub") if is_navigational_format?
```

Reply:
We can do this, but decided to hold off on it as we need to override the Devise failure app as well. It doesn't allow customizing which flash keys to use.

https://github.com/heartcombo/devise/blob/master/lib/devise/failure_app.rb